ui-tests/src/test/java/io/syndesis/qe/steps/other/ConditionalFlowSteps.java

```diff
@@ -0,0 +1,164 @@
+
+import com.codeborne.selenide.ElementsCollection;
+import com.codeborne.selenide.SelenideElement;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import io.cucumber.datatable.DataTable;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ConditionalFlowSteps {
+    private static class EditFlowStepElements {
+        private static By ICON_DOWN = By.className("fa-arrow-circle-o-down");
+        private static By ICON_UP = By.className("fa-arrow-circle-o-up");
+        private static By ICON_DELETE = By.className("fa-trash-o");
+        private static By CONDITION_LIST_ITEM = By.className("form-control");
+        private static By OPEN_FLOW_CONDITION_BUTTON_PARENT = By.className("list-view-pf-actions");
+        private static By FLOW_DROPDOWN = By.className("pf-c-dropdown__toggle-icon");
+        private static By FLOW_DROPDOWN_ITEM = By.className("pf-c-dropdown__menu-item");
+
+        private static By getConditionOnPosition(String position) {
+            return By.cssSelector("[data-testid=\"flowconditions-X-condition\"]".replace("X", position));
+        }
+
+        private static By getConditionIconsOnPosition(String position) {
+            return By.id("flowconditions-X-array-controls-control".replace("X", position));
+        }
+    }
+
+    private static class EditIntegrationElements {
+        private static By CONDITIONAL_FLOW_STEP = By.cssSelector("[data-testid=\"integration-editor-steps-list-item-conditional-flows-list-item\"]");
+        private static By CONDITIONAL_FLOW_STEP_INNER_FLOW_ITEM = By.className("list-group-item");
+        private static By DEFAULT_DISABLED = By.className("fa-ban");
+    }
+
+    //data:     |position   |value  |
+    //example:  |0          |asdf   |
+    @Then("^validate condition content in condition flow step$")
+    public void validateConditionContent(DataTable data) {
+
+        for (List<String> row : data.cells()) {
+            assertThat($(EditFlowStepElements.getConditionOnPosition(row.get(0))).getValue())
+                .isEqualToIgnoringCase(row.get(1));
+
+            //selenide returns empty value in some cases even if the value is present, probably a driver bug
+            // so second assert with usage of javascript is necessary
+            assertThat((String) executeJavaScript("return document.getElementById('flowconditions-" + row.get(0) + "-condition').value"))
+                .isEqualToIgnoringCase(row.get(1));
+        }
+    }
+
+    //data:
+    //  |position   |ACTION   |
+    //  |1          |UP       |
+    //  |0          |DOWN     |
+    //  |1          |DELETE   |
+    @When("^click on the condition icon$")
+    public void sendEmail(DataTable data) {
+        for (List<String> row : data.cells()) {
+            switch (row.get(1)) {
+                case "UP":
+                    $(EditFlowStepElements.getConditionIconsOnPosition(row.get(0)))
```

Comment:
Having just the `EditFlowStepElements.ICON_FOO` assignment inside the switch statement and the rest of the call outside will make this easier to refactor/fix in the future